src/sidebar/app/components/Footer.js

```diff
@@ -0,0 +1,333 @@
+        // disconnectSync.style.display = 'none';
+        // footerButtons.title = null; // remove profile email from title attribute
+        this.setState({
+          isAuthenticated: false
+        });
+        this.getLastSyncedTime();
+        break;
+    }
+  };
+
+  getLastSyncedTime = () => {
+    if (this.state.waitingToReconnect) {
+      // persist reconnect warning, do not override with the 'saved at'
+      return;
+    }
+
+    if (this.state.isAuthenticated) {
+
+      this.setState({
+        savingIndicatorText: browser.i18n.getMessage('syncComplete2', formatFooterTime(this.state.lastModified)),
+        isAuthenticated: true
+      });
+      setAnimation(false, true);
+      // disconnectSync.style.display = 'block';
+    } else {
+      this.setState({
+        savingIndicatorText: browser.i18n.getMessage('savedComplete2', formatFooterTime(this.state.lastModified))
+      });
+    }
+  };
+
+
+  disconnectFromSync = () => {
+    this.setState({
+      waitingToReconnect: false,
+      isAuthenticated: false
+    });
+    // disconnectSync.style.display = 'none';
+    setAnimation(false, false, false); // animateSyncIcon, syncingLayout, warning
+    setTimeout(() => {
+      this.setState({
+        savingIndicatorText: browser.i18n.getMessage('disconnected')
+      });
+    }, 200);
+    setTimeout(() => {
+      this.getLastSyncedTime();
+    }, 2000);
+    browser.runtime.sendMessage('notes@mozilla.com', {
+      action: 'disconnected'
+    });
+  };
+
+  enableSyncAction = (editor) => {
+    if (this.state.editingInProcess || this.state.syncingInProcess) {
+      return;
+    }
+
+    const footerButtons = document.getElementById('footer-buttons');
+
+    if (this.state.isAuthenticated && footerButtons.classList.contains('syncingLayout')) {
+      // Trigger manual sync
+      setAnimation(true);
+      browser.runtime.sendMessage({
+          action: 'kinto-sync'
+        });
+    } else if (!this.state.isAuthenticated && (footerButtons.classList.contains('savingLayout') || this.state.waitingToReconnect)) {
+      // Login
+      setAnimation(true, true, false);  // animateSyncIcon, syncingLayout, warning
+
+      // enable disable sync button
+      // disconnectSync.style.display = 'block';
+
+      setTimeout(() => {
+        this.setState({
+          savingIndicatorText: browser.i18n.getMessage('openingLogin')
+        });
+      }, 200); // Delay text for smooth animation
+
+      this.loginTimeout = setTimeout(() => {
+        setAnimation(false, true, true); // animateSyncIcon, syncingLayout, warning
+        this.setState({
+          savingIndicatorText: browser.i18n.getMessage('pleaseLogin'),
+          waitingToReconnect: true
+        });
+      }, 5000);
+
+      // Problem not having editor in Footer Component
+      browser.runtime.sendMessage({
+        action: 'authenticate',
+        context: null // getPadStats(editor)
+      });
+    }
+
+    this.setState({
+      waitingToReconnect: false
```

Comment:
what's the plan to share "connected" state between the Header and Footer? (in the new design the disable sync button is in the Header, while the connection state is in the footer) 🤔

Reply:
Our plan is to integrate a Redux interface to keep state in sync between all components. First I will finish migration, than start refactoring with some React sugar syntax, and finally implement redux. 😊